src/main/java/hudson/plugins/git/extensions/BrowserInference.java

```diff
@@ -0,0 +1,22 @@
+package hudson.plugins.git.extensions;
+
+import hudson.plugins.git.browser.GitRepositoryBrowser;
+import hudson.scm.RepositoryBrowsers;
+import org.apache.tools.ant.ExtensionPoint;
+
+public abstract class BrowserInference extends ExtensionPoint {
```

Comment:
Here you should wonder:
1. Why abstract?
1. Will you allow the rest of plugins to consume this extension? Or is it something that will be part only of the git plugin?
1. Is some sort of UI needed? Will you offer a new UI?

If this extension is something that will never extended by other plugins, then:
1. Annotate it as `@Restricted(NoExternalUse.class)` as in https://github.com/jenkinsci/git-plugin/blob/c0e8144da63e3c8ccea7f00f7438b6e70aac9b00/src/main/java/jenkins/plugins/git/traits/GitBrowserSCMSourceTrait.java#L110
1. There is no need to make it abstract. Use abstract classes when you want them to be extended.

If there is some expected offered UI, you should implement the Descriptor. See the `Describable/Descriptor pattern` in  https://wiki.jenkins.io/display/JENKINS/Defining+a+new+extension+point